Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionCiscoASA.yml

```diff
@@ -0,0 +1,369 @@
+      let all_106001_alike = allLogs
+                | where DeviceEventClassID in ("106001", "106006", "106015", "106016", "106021", "106022") 
+                | parse Message with * " interface " DstInterfaceName;
+      let all_106010_alike = allLogs
+                | where DeviceEventClassID in ("106010", "106014")
+                | parse Message with * " src "  SrcInterfaceName ":" * " dst " DstInterfaceName ":" * "(type " NetworkIcmpType ", code " NetworkIcmpCode:int ")";
+      let all_106018 = allLogs
+                | where DeviceEventClassID == "106018"
+                | parse Message with * " packet type " NetworkIcmpType " " * "list " NetworkRuleName " " *;
+      let all_106023 = allLogs
+                | where DeviceEventClassID == "106023" and not(Message has "protocol 41") and isnotempty(SourceIP)
+                | parse Message with * " src "  SrcInterfaceName ":" * " dst " DstInterfaceName ":" * ' by access-group "' NetworkRuleName '" '  *
+                | parse Message with * "(type " NetworkIcmpType ", code " NetworkIcmpCode:int ")" *;
+      let all_106023_unparsed = allLogs
+                | where DeviceEventClassID == "106023" and isempty(SourceIP) and not(Message has "protocol 41")
+                | parse Message with * ": " DeviceAction " " Protocol " src " SrcInterfaceName ":" SrcIpAddrAndPort "(" SrcUsername ") dst " DstInterfaceName ":" DstIpAddrAndPort " " NetworkIcmpInfo 'by access-group "' NetworkRuleName '" [' * "]"
+                | parse NetworkIcmpInfo with "(type " NetworkIcmpType ", code " NetworkIcmpCode:int ") "
+                | extend SrcIpAddr = tostring(split(SrcIpAddrAndPort,"/",0)[0]),
+                        SrcPortNumber = toint(split(SrcIpAddrAndPort,"/",1)[0]),
+                        DstIpAddr = tostring(split(DstIpAddrAndPort,"/",0)[0]),
+                        DstPortNumber = toint(split(DstIpAddrAndPort,"/",1)[0])
+                | extend 
+                        SrcUsernameType = iif(isnotempty(SrcUsername),"Windows",""),
```

Comment:
Is the username, here and in other events, actually windows format (i.e. domain\user)?

Reply:
Some examples of Username values are LOCAL\U1613@ABC.COM, U1613@ABC.COM, U1613, 512:XYZ_U1632_Group1

Reply:
So, identifying type is harder and some values don't really follow types I know (the 512: one and the one with both @ and \):
I would do with "@" as UPN, with "\" (if there is no "@") as Windows, and Simple otherwise (U1613).

Reply:
Updated